protocol/river-control-unstable-v1.xml

```diff
@@ -37,49 +29,421 @@
+    </request>
+
+    <request name="mod_main_count">
+      <description summary="modify number of main views">
+        Modify the number of main views of the focused output.
+      </description>
+      <arg name="seat" type="object" interface="wl_seat"/>
+      <arg name="delta" type="int" summary="delta applied to the count"/>
+    </request>
+
+    <request name="mod_main_factor">
+      <description summary="modify the size of the main area">
+        The main area is described by a value between 0.0 and 1.0 where 1.0.
+        Layout generators may use this in various ways. For the default
+        rivertile layout generator, this value is interpreted as a percentage
+        of the total width of the output.
+      </description>
+      <arg name="seat" type="object" interface="wl_seat"/>
+      <arg name="delta" type="fixed" summary="delta applied to the factor"/>
+    </request>
+
+    <enum name="axis">
+      <entry name="horizontal" value="0"/>
+      <entry name="vertical" value="1"/>
+    </enum>
+
+    <request name="move">
+      <description summary="move the focused view">
+        The view is set to floating if not already floating.
+      </description>
+      <arg name="seat" type="object" interface="wl_seat"/>
+      <arg name="axis" type="uint" enum="axis" summary="axis to move along"/>
+      <arg name="delta" type="int" summary="amount/direction along the axis"/>
+    </request>
+
+    <request name="resize">
+      <description summary="resize the focused view">
+        The view is set to floating if not already floating.
+      </description>
+      <arg name="seat" type="object" interface="wl_seat"/>
+      <arg name="axis" type="uint" enum="axis" summary="axis to resize along"/>
+      <arg name="delta" type="int" summary="amount/direction along the axis"/>
+    </request>
+
+    <enum name="edge">
+      <entry name="top" value="0"/>
+      <entry name="bottom" value="1"/>
+      <entry name="left" value="2"/>
+      <entry name="right" value="3"/>
+    </enum>
+
+    <request name="snap">
+      <description summary="snap the focused view to a screen edge">
+        The view is set to floating if not already floating.
+      </description>
+      <arg name="seat" type="object" interface="wl_seat"/>
+      <arg name="edge" type="uint" enum="edge" summary="edge to snap to"/>
+    </request>
+
+    <request name="layout">
+      <description summary="set the layout command">
+        Provide a command which river will use for generating the layout
+        of non-floating windows on the currently focused output. See
+        *river-layouts*(7) for details on the expected formatting of the
+        output of layout commands. Alternatively, "full" can be given
+        instead of a command to cause river to use its single internal layout,
+        in which windows span the entire width and height of the output.
+      </description>
+      <arg name="seat" type="object" interface="wl_seat"/>
```

Comment:
Instead of getting the focused output of the seat, maybe this request should just have an `output` parameter. A client can get the focused output with `river-status-v1`.

Reply:
> Although if we want river to be really scriptable then in the long run we may want to think about a protocol that allows operations on arbitrary views, not just the focused one.

You make a good point, this current draft is still very close to a direct translation of the previous command interface. We do have an opportunity to make this a lot more flexible while still exposing the same command line interface through riverctl. I'll have to give some more thought towards how to best handle addressing of views.